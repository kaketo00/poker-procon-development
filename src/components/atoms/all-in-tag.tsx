import { Tag } from '@chakra-ui/react';

const AllInTag = () => (
  <Tag bgColor="brand100.300" variant="solid" size="sm">
    ALL-IN
  </Tag>
);

export default AllInTag;
