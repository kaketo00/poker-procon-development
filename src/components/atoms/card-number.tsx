import type { TextProps } from '@chakra-ui/react';
import { Flex, Text } from '@chakra-ui/react';

import type { Size } from '@/schema/style';

import { SUITS } from '../../constants/index';
import { ClubIcon, DiamondIcon, HeartIcon, SpadeIcon } from './icon';

export interface CardNumberProps extends TextProps {
  number: number | null;
  size: Size;
  reverse?: boolean;
  index: number;
}

const CardNumber = ({
  number,
  size,
  reverse,
  index,
  ...rest
}: CardNumberProps) => {
  const suit = SUITS[index % 4];

  return (
    <Flex
      position="absolute"
      transform={reverse ? 'scale(-1, -1)' : 'scale(1, 1)'}
      justify="center"
      align="center"
      w={size === 'sm' ? 6 : 10}
      h={size === 'sm' ? 6 : 10}
      {...rest}
    >
      <Text
        zIndex="docked"
        color="white"
        mb={0.5}
        textStyle={size === 'sm' ? 'bold.xs' : 'bold.md'}
        textShadow={`0 1px ${
          suit === 'Spades' || suit === 'Clubs'
            ? 'black'
            : 'var(--chakra-colors-brand100-300)'
        }`}
      >
        {number}
      </Text>
      {suit === 'Clubs' && <ClubIcon size={size} />}
      {suit === 'Diamonds' && <DiamondIcon size={size} />}
      {suit === 'Hearts' && <HeartIcon size={size} />}
      {suit === 'Spades' && <SpadeIcon size={size} />}
    </Flex>
  );
};

export default CardNumber;
