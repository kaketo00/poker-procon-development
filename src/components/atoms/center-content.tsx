import type { FlexProps } from '@chakra-ui/react';
import { Box } from '@chakra-ui/react';

const CenterContent = ({ children }: FlexProps) => (
  <Box
    position="absolute"
    top="50%"
    left="50%"
    transform="translate(-50%, -50%)"
    maxW="5xl"
    width="full"
  >
    {children}
  </Box>
);

export default CenterContent;
