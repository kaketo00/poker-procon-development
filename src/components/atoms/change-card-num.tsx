import { Flex, Text } from '@chakra-ui/react';

import type { ChangeNum, ChangeTimes } from '@/schema/style';

import ChangeBgFirst from './changes/change-bg-1st';
import ChangeBgSecond from './changes/change-bg-2nd';

export type ChangeCardNumProps = {
  num: ChangeNum;
  times: ChangeTimes;
};

const ChangeCardNum = ({ num, times }: ChangeCardNumProps) => (
  <Flex justify="center" align="center" w={6} h={6}>
    {times === 1 && (
      <>
        <ChangeBgFirst />
        <Text position="absolute" textStyle="bold.xs" textAlign="center">
          {num}
        </Text>
      </>
    )}
    {times === 2 && (
      <>
        <ChangeBgSecond />
        <Text position="absolute" textStyle="bold.xs" textAlign="center">
          {num}
        </Text>
      </>
    )}
  </Flex>
);

export default ChangeCardNum;
