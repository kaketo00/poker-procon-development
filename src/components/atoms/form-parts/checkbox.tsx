import type { CheckboxProps as ChakraCheckboxProps } from '@chakra-ui/react';
import { Checkbox as ChakraCheckbox } from '@chakra-ui/react';
import { forwardRef } from 'react';

interface CheckboxProps extends ChakraCheckboxProps {
  label: string;
}

const Checkbox = forwardRef<HTMLElement, CheckboxProps>(
  ({ label, ...rest }, ref) => (
    <ChakraCheckbox
      iconColor="brand100.100"
      {...rest}
      _checked={{
        '& .chakra-checkbox__control': {
          background: 'white',
          borderColor: 'inherit',
        },
      }}
      ref={ref}
    >
      {label}
    </ChakraCheckbox>
  )
);

Checkbox.displayName = 'Checkbox';

export default Checkbox;
