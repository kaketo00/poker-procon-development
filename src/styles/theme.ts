import { extendTheme } from '@chakra-ui/react';
import { Zen_Kaku_Gothic_New } from 'next/font/google';

const nextFont = Zen_Kaku_Gothic_New({
  weight: ['400', '700', '900'],
  subsets: ['latin'],
});

const theme = extendTheme({
  fonts: {
    heading: nextFont.style.fontFamily,
    body: nextFont.style.fontFamily,
  },
  colors: {
    brand100: {
      100: '#0DD203', // green 100%
      200: '#005CFF', // blue 100%
      300: '#B4171E', // red 100%
      600: '#777777', // dark-gray 100%
      700: '#D4D4D4', // gray 100%
      800: '#ECECEC', // light-gray 100%
    },
    brand80: {
      100: 'rgba(13, 210, 3, 0.8)', // green 80%
      200: 'rgba(0, 92, 255, 0.8)', // blue 80%
      300: 'rgba(180, 23, 30, 0.8)', // red 80%
      400: 'rgba(255, 255, 255, 0.8)', // white 80%
      500: 'rgba(0, 0, 0, 0.8)', // black 80%
      600: 'rgba(119, 119, 119, 0.8)', // gray 80%
    },
  },
  textStyles: {
    normal: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'normal',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'normal',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'normal',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'normal',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'normal',
        lineHeight: '1.75rem',
      },
    },
    bold: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'bold',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'bold',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'bold',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'bold',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'bold',
        lineHeight: '1.75rem',
      },
    },
    strong: {
      xs: {
        fontSize: '0.6875rem',
        fontWeight: 'black',
        lineHeight: '1rem',
      },
      sm: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      md: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '2rem',
      },
      lg: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.375rem',
      },
      xl: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
    },
    heading: {
      h6: {
        fontSize: '1rem',
        fontWeight: 'black',
        lineHeight: '1.25rem',
      },
      h5: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      h4: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '1.875rem',
      },
      h3: {
        fontSize: '1.75rem',
        fontWeight: 'black',
        lineHeight: '2.125rem',
      },
      h2: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
      h1: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.25rem',
      },
    },
    shadows: {
      light5: {
        100: '0px 0px 5px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 5px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 5px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 5px rgba(255, 255, 255, 0.8)',
      },
      light10: {
        100: '0px 0px 10px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 10px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 10px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 10px rgba(255, 255, 255, 0.8)',
      },
    },
  },
  breakpoints: {
    base: '0px', // 0 - 1279px game-template(S)
    sm: '80em', // 1280px - 1439px game-template(S)
    md: '90em', // 1440px - 1919px game-template(M)
    lg: '120em', // 1920px - 2559px game-template(M)
    xl: '160em', // 2560px - âˆž game-template(L)
    '2xl': '160em',
  },
});

export default theme;
