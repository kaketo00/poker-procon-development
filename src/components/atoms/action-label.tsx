import type { BoxProps } from '@chakra-ui/react';
import { Box, Text } from '@chakra-ui/react';
import { useMemo } from 'react';

import type { Action } from '@/schema/game';
import type { PlayerStatus } from '@/schema/player';
import { strongM } from '@/styles';
import { capitalize } from '@/utils/style';

export interface ActionLabelProps extends BoxProps {
  label: Action | PlayerStatus | string;
  add: number;
}
const ActionLabel = ({ label, add, ...rest }: ActionLabelProps) => {
  const color = useMemo(() => {
    switch (label) {
      case 'bet':
      case 'raise':
        return 'brand80.100';
      case 'check':
      case 'call':
        return 'brand80.200';
      case 'all-in':
        return 'brand80.300';
      case 'drop':
      case 'out':
        return 'brand80.600';
      default:
        return 'brand80.400';
    }
  }, [label]);

  const shadowColor = useMemo(() => {
    switch (label) {
      case 'bet':
      case 'raise':
        return '0 0 10px rgba(10, 210, 0, 0.8)';
      case 'check':
      case 'call':
        return '0 0 10px rgba(0, 92, 255, 0.8)';
      case 'all-in':
        return '0 0 10px rgba(180, 23, 30, 0.8)';
      case 'drop':
      case 'out':
        return 'none';
      default:
        return '0 0 10px rgba(255, 255, 255, 0.8)';
    }
  }, [label]);

  const viewLabel = useMemo(() => {
    if (!label) return '';
    if (label === 'bet' || label === 'raise')
      return `${capitalize(label)} +${add.toLocaleString()}pt`;

    return capitalize(label);
  }, [label]);

  return (
    <Box minH={['28px', '28px', '36px', '42px', '42px', '42px']} {...rest}>
      {label && (
        <Text
          textStyle={strongM}
          display="block"
          px={5}
          bgColor="brand80.500"
          borderColor={color}
          borderWidth={2}
          borderRadius="base"
          boxShadow={shadowColor}
          color={color}
          fontWeight="bold"
        >
          {viewLabel}
        </Text>
      )}
    </Box>
  );
};

export default ActionLabel;
