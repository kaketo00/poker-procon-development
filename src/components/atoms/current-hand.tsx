import type { BoxProps } from '@chakra-ui/react';
import { Box, Image, Text } from '@chakra-ui/react';
import { useMemo } from 'react';

import { HAND_RANK } from '@/constants/game';
import type { Card, CardSet } from '@/schema/card';
import type { Phase } from '@/schema/game';
import { normalM } from '@/styles';
import { evaluateHand } from '@/utils/hand';

export interface CurrentHandProps extends BoxProps {
  phase: Phase;
  beforeCards?: CardSet<Card>;
  afterCards: CardSet<Card>;
}

const CurrentHand = ({
  phase,
  beforeCards,
  afterCards,
  ...rest
}: CurrentHandProps) => {
  const beforeHand = useMemo(
    () => (beforeCards ? evaluateHand(beforeCards) : ''),
    [beforeCards]
  );

  const afterHand = useMemo(() => evaluateHand(afterCards), [afterCards]);

  if (phase === 'finished' || !afterCards.length) return null;

  return (
    <Box
      minH={['28px', '28px', '36px', '42px', '42px', '42px']}
      w="max-content"
      {...rest}
    >
      <Text
        textStyle={normalM}
        display="block"
        py={0.5}
        px={5}
        bgColor="brand80.500"
        borderRadius="base"
        color="white"
      >
        {beforeHand ? `${HAND_RANK[beforeHand]} â†’ ` : ''}
        {HAND_RANK[afterHand]}
      </Text>
      <Image src="/bubble.svg" alt="bubble" mt={['px', 'px', 0.5, 1]} />
    </Box>
  );
};

export default CurrentHand;
