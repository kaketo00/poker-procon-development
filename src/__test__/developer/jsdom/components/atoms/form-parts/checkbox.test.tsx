import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';

import type { CheckboxProps } from '@/components/atoms/form-parts';
import { Checkbox } from '@/components/atoms/form-parts';
import theme from '@/styles/theme';

const Component = (props: CheckboxProps) => (
  <ChakraProvider theme={theme}>
    <Checkbox {...props} />
  </ChakraProvider>
);

describe('CheckBox', () => {
  test('スナップショット/default', () => {
    const { asFragment } = render(<Component label="Player1" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
