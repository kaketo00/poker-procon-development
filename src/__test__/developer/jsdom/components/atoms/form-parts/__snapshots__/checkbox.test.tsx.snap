// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox スナップショット/default 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-ctlur2"
  >
    <input
      class="chakra-checkbox__input"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-1ydjfm6"
    />
    <span
      class="chakra-checkbox__label css-6x44c9"
    >
      Player1
    </span>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
